neal/sampler.py

```diff
@@ -305,20 +302,28 @@ def _default_ising_beta_range(h, J):
 
     Assume each variable in J is also in h.
 
+    We shoot for a final normalized beta of 10. We use the minimum bias to
+    give a lower bound on the minimum energy gap, such at the final sweeps
+    we are highly likely to settle into the current valley.
     """
+
     bias_sum = 0.0
     sigmas = {}
+    min_bias = float("inf")
     for v, b in iteritems(h):
         bias_sum += b**2
-        sigmas[v] = abs(b)
+        abs_b = abs(b)
+        sigmas[v] = abs_b
+        min_bias = min(abs_b if abs_b != 0 else float("inf"), min_bias)
```

Comment:
Assuming `abs_b` is always a number (or also ignored if `None`), shortcircuiting works too:
```
min_bias = min(abs_b or float("inf"), min_bias)
```
:grin: